src/main/java/io/frictionlessdata/datapackage/JSONBase.java

```diff
@@ -165,63 +169,63 @@ public String getSchemaReference() {
         Object dialectObj = resourceJson.has(JSONBase.JSON_KEY_DIALECT) ? resourceJson.get(JSONBase.JSON_KEY_DIALECT) : null;
-        JSONObject dereferencedDialect = dereference(dialectObj, basePath, isArchivePackage);
+        JsonNode dereferencedDialect = dereference(dialectObj, basePath, isArchivePackage);
         if (null != dereferencedDialect) {
             return Dialect.fromJson(dereferencedDialect.toString());
         }
         return null;
     }
 
-    public static void setFromJson(JSONObject resourceJson, JSONBase retVal, Schema schema) {
-        if (resourceJson.has(JSONBase.JSON_KEY_SCHEMA))
-            retVal.originalReferences.put(JSONBase.JSON_KEY_SCHEMA, resourceJson.get(JSONBase.JSON_KEY_SCHEMA));
-        if (resourceJson.has(JSONBase.JSON_KEY_DIALECT))
-            retVal.originalReferences.put(JSONBase.JSON_KEY_DIALECT, resourceJson.get(JSONBase.JSON_KEY_DIALECT));
-
-        //FIXME: Again, could be greatly simplified amd much more
-        // elegant if we use a library like GJSON...
-        String name = resourceJson.has(JSONBase.JSON_KEY_NAME) ? resourceJson.getString(JSONBase.JSON_KEY_NAME) : null;
-        String profile = resourceJson.has(JSONBase.JSON_KEY_PROFILE) ? resourceJson.getString(JSONBase.JSON_KEY_PROFILE) : null;
-        String title = resourceJson.has(JSONBase.JSON_KEY_TITLE) ? resourceJson.getString(JSONBase.JSON_KEY_TITLE) : null;
-        String description = resourceJson.has(JSONBase.JSON_KEY_DESCRIPTION) ? resourceJson.getString(JSONBase.JSON_KEY_DESCRIPTION) : null;
-        String mediaType = resourceJson.has(JSONBase.JSON_KEY_MEDIA_TYPE) ? resourceJson.getString(JSONBase.JSON_KEY_MEDIA_TYPE) : null;
-        String encoding = resourceJson.has(JSONBase.JSON_KEY_ENCODING) ? resourceJson.getString(JSONBase.JSON_KEY_ENCODING) : null;
-        Integer bytes = resourceJson.has(JSONBase.JSON_KEY_BYTES) ? resourceJson.getInt(JSONBase.JSON_KEY_BYTES) : null;
-        String hash = resourceJson.has(JSONBase.JSON_KEY_HASH) ? resourceJson.getString(JSONBase.JSON_KEY_HASH) : null;
-
-        JSONArray sources = resourceJson.has(JSONBase.JSON_KEY_SOURCES) ? resourceJson.getJSONArray(JSONBase.JSON_KEY_SOURCES) : null;
-        JSONArray licenses = resourceJson.has(JSONBase.JSON_KEY_LICENSES) ? resourceJson.getJSONArray(JSONBase.JSON_KEY_LICENSES) : null;
+    public static void setFromJson(JsonNode resourceJson, JSONBase retVal, Schema schema) {
+    	
+        if (resourceJson.has(JSONBase.JSON_KEY_SCHEMA) && resourceJson.get(JSONBase.JSON_KEY_SCHEMA).isTextual())
+            retVal.originalReferences.put(JSONBase.JSON_KEY_SCHEMA, resourceJson.get(JSONBase.JSON_KEY_SCHEMA).asText());
+        if (resourceJson.has(JSONBase.JSON_KEY_DIALECT) && resourceJson.get(JSONBase.JSON_KEY_DIALECT).isTextual())
+            retVal.originalReferences.put(JSONBase.JSON_KEY_DIALECT, resourceJson.get(JSONBase.JSON_KEY_DIALECT).asText());
+
+        // TODO: A mapper library might be useful, but not required
```

Comment:
Clean code is indeed never *required* strictly speaking, but this is really crying out to be replaced by standard Jackson bean deserialization.

Reply:
Agreed - I think all JsonNode parameters and return values should be removed.